lib/ArduinoNative/src/WString.h

```diff
@@ -466,6 +466,44 @@ class String
         return isEmptyFlag;
     }
 
+    /**
+     * Remove all characters from the String tarting at the index until the end of the String.
+     * @param[in] index Index at which to remove characters.
+     */
+    void remote(unsigned int index)
+    {
+        remove(index, (unsigned int)-1);
+    }
+
+    /**
+     * Remove a number of characters from the String starting at the index.
+     * @param[in] index Index at which to remove characters.
+     * @param[in] count Number of characters to remove.
+     */
+    void remove(unsigned int index, unsigned int count)
+    {
+        if (index >= length())
+        {
+            return;
+        }
+        if (count <= 0)
+        {
+            return;
+        }
+        if (count > length() - index)
+        {
+            count = length() - index;
+        }
+        unsigned int newlen = length() - count;
+        char*        tmp    = new char[newlen + 1];
+        memcpy(tmp, m_buffer, index);

```

Comment:
Did you get the buffer?

Reply:
Sorry, I do not understand the question

Reply:
NULL pointer check is missing, you can not be sure that you got the buffer.